app/models/comment.rb

```diff
@@ -214,6 +218,10 @@ def synchronous_bust
     expire_root_fragment
   end
 
+  def root_exists?
+    ancestry && Comment.find_by(id: ancestry)
```

Comment:
Since we don't need to use the comment object, we can use `Comment.exists?(id: ancestry)`